BootloaderCorePkg/Stage1A/Stage1A.c

```diff
@@ -359,15 +360,30 @@ SecStartup2 (
     SetLibraryData (PcdGet8 (PcdPcdLibId), LdrGlobal->PcdDataPtr, BufInfo->AllocLen);
   }
 
-  // Extra initialization
-  if (FlashMap != NULL) {
-    SetCurrentBootPartition ((FlashMap->Attributes & FLASH_MAP_ATTRIBUTES_BACKUP_REGION) ? 1 : 0);
+  if (PcdGetBool (PcdIdenticalTopSwapsBuilt)) {
+    Status = GetBootPartition (&Partition);
+    if (EFI_ERROR (Status)) {
+      CpuHalt ("Boot partition retrieval failed!\n");
+    }
+    SetCurrentBootPartition (Partition);
+  } else {
+    if (FlashMap != NULL) {
+      SetCurrentBootPartition ((FlashMap->Attributes & FLASH_MAP_ATTRIBUTES_BACKUP_REGION) ? 1 : 0);
+    } else {
+      CpuHalt ("Boot partition retrival failed!\n");
```

Comment:
Should we actually do this?

Reply:
Further down in this function, the CPU gets halted if the flash map is null. In the case where we can't get the BP from the TS register, I default the partition to 0, to avoid another CPU halt.